### ISSUE TITLE
Using hotplate to heat items sometimes does not result in hot food

### ISSUE THREAD

**Issue by failboat78** [CleverRaven/Cataclysm-DDA#76413]
### Describe the bug

Not sure why this isn't working sometimes. Please load the save and try to reproduce. 

Maybe it's when the food is coming from the fridge and is already cold?

Try to use the hotplate to heat a toastem. I note that it doesn't seem possible to heat the item a second item - even tho it still says it's cold. Display bug maybe?

This worked fine in the past for me.

### Attach save file

[Sunnydale-trimmed.tar.gz](https://github.com/user-attachments/files/16995684/Sunnydale-trimmed.tar.gz)

### Steps to reproduce

Use the nearby hotplate to heat an item
Observe after whether or not hot item is available

### Expected behavior

Expect to have hot food after heating it

### Screenshots

![image](https://github.com/user-attachments/assets/083ecc51-3063-4c91-bb2f-3c4043af7cbe)

### Versions and configuration

- OS: Linux
    - OS Version: <unknown>
- Game Version: cdda-experimental-2024-09-12-1816 2473620 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mythical Martial Arts [MMA],
    Bombastic Perks [bombastic_perks],
    Mind Over Matter [mindovermatter]
]

**Comment by PatrikLundell** [comment]
Failed to repeat.
I never managed to find the "nearby" hotplate, nor any toast-em (it seams to have been eaten).
Thus, I spawned a new toast-em (which spawns at ambient temperature) and a hotplate that was connected to the oven.
Heated the toast-em with the hotplate and the toast-em became hot.

The history indicated the "nearby" hotplate had been modified, so it might work differently from a regular one.

**Comment by failboat78** [comment]
There is an oven (with hotplate) right there in the kitchen. There are heatable items in the fridge and the cupboard 2 tiles away.

**Comment by PatrikLundell** [comment]
/Confirmed

It's better to describe what you actually tried to do, rather than the generic action taken, as it makes it possible to replicate the actual action. It's not unusual that the specific sequence is needed for replication.

I was eventually able to replicate your issue. The oven didn't show up in the activation list, and the 'e'xamine UI is broken such that the line for heating up items is hidden below the bottom edge of the menu, but activating the hidden line brought up the heating interaction (another thing among many where a recent change has caused trouble with the UI). The toast-em from the fridge did indeed remain cold after being "heated" by the oven.

**Comment by failboat78** [comment]
Yes I should have detailed that, sorry. This was a weird one - sometimes I was heating stuff and then heating hot items. But other times not - so there was some head-scratching...

**Comment by failboat78** [comment]
It's not just the fridge causing this. I get same problem if I heat items from the cupboard. The items do say (cold) tho before and after heating.

**Comment by failboat78** [comment]
@PatrikLundell Some more info from testing which may be helpful:
Using hotplate to heat item in players inventory = works as expected
Using hotplate to heat item that is not in players inventory (ie is in fridge or cupboard) = does not result in hot item

**Comment by AideeAmei** [comment]
I hereby confirm this, attaching a save with no mods. Induction hotplate (seems to be the same as hotplate in this matter) in inventory is attached to vehicle controls to the west, onion (cold) is on the adjacent tile to the west.

Version: 512963f (tiles)
Build: 2024-12-04-0642
OS: Windows 10 Pro 22H2 19045.5131

[CleanFreshTestingWorldNoMods01.zip](https://github.com/user-attachments/files/18008380/CleanFreshTestingWorldNoMods01.zip)

**Comment by AideeAmei** [comment]
Sorry, you have to test it with some other stuff in the vehicle. The onion, despite marked (cold), isn't in the list of items to heat any more (I tested it with that onion and then saved). There's a fridge full of cold stuff in the back.

**Comment by molkemon** [comment]
Also experiencing this (tried with an oven hotplate aswell as a microwave plugged into a vehicle kitchen). As others have pointed out you can't heat it a second time, but more importantly, the item, despite saying it's cold is ACTUALLY HOT. If you eat it, you will get morale bonus from hot meal.

**Comment by AideeAmei** [comment]
> despite saying it's cold is ACTUALLY HOT. If you eat it, you will get morale bonus from hot meal.

Thanks for this!
